### PR TITLE
Only allow session auth to be used for a single auth flow

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

**Comment by liggitt** [comment]
@smarterclayton travis is green, it's a sign

**Comment by liggitt** [comment]
[test]

**Comment by liggitt** [comment]
@deads take a quick look?

File: pkg/auth/server/session/authenticator.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func (a *Authenticator) AuthenticationSucceeded(user api.UserInfo, state string,
 	return false, a.store.Save(w, req)
 }
 
-func (a *Authenticator) InvalidateAuthentication(context api.UserInfo, w http.ResponseWriter, req *http.Request) error {
+func (a *Authenticator) InvalidateAuthentication(w http.ResponseWriter, req *http.Request) error {
 	session, err := a.store.Get(req, a.name)
```
Comment (deads2k) [inline pkg/auth/server/session/authenticator.go]: This always returns an error instead of a nil?

**Comment by deads2k** [comment]
One question and a personal distaste of function pointers as types (recognized as a personal problem).  Otherwise, lgtm.

File: pkg/auth/server/session/authenticator.go (hunk 1 above)
Comment (liggitt) [inline pkg/auth/server/session/authenticator.go]: ?

Comment (liggitt) [inline pkg/auth/server/session/authenticator.go]: Yes, it always returns a session or an err

**Comment by deads2k** [comment]
lgtm [merge]

**Comment by liggitt** [comment]
[merge]

**Pull request merged**